app/controllers/order/goods_details.js

```diff
@@ -1,82 +1,99 @@
+      orderId,
+      params
+    );
+    const goodcityRequests = this.get("goodcityRequests");
+    goodcityRequests[index].id = data["goodcity_request"]["id"];
+    this.set("goodcityRequests", [...goodcityRequests]);
+  },
+
+  async updateGoodsDetails(orderId, params) {
+    await this.get("orderService").updateGoodsDetails(orderId, params);
+  },
+
   actions: {
     addRequest() {
-      var orderId = this.get("order.id");
-      var goodcityRequestParams = {};
-      goodcityRequestParams["quantity"] = 1;
-      goodcityRequestParams["order_id"] = orderId;
-      var loadingView = getOwner(this)
-        .lookup("component:loading")
-        .append();
+      const goodcityRequest = {
+        description: "",
+        quantity: 1,
+        packageType: null
+      };
+      this.set("goodcityRequests", [
+        ...this.get("goodcityRequests"),
+        goodcityRequest
+      ]);
+    },
 
-      new AjaxPromise(
-        "/goodcity_requests",
-        "POST",
-        this.get("session.authToken"),
-        { goodcity_request: goodcityRequestParams }
-      )
-        .then(data => {
-          this.get("store").pushPayload(data);
-        })
-        .finally(() => {
-          loadingView.destroy();
-        });
+    async onRemoveRequest(_id, index) {
+      this.set("goodcityRequests", [
+        ...this.get("goodcityRequests").slice(0, index),
+        ...this.get("goodcityRequests").slice(index + 1)
+      ]);
     },
 
-    saveGoodsDetails() {
+    async saveGoodsDetails() {
       if (this.get("hasNoGcRequests")) {
         return false;
       }
-      var promises = [];
-      this.get("order.goodcityRequests").forEach(goodcityRequest => {
-        promises.push(goodcityRequest.save());
-      });
-
-      var loadingView = getOwner(this)
-        .lookup("component:loading")
-        .append();
-
-      all(promises)
-        .then(function() {
-          loadingView.destroy();
-        })
-        .finally(() => {
-          this.transitionToRoute(
-            "order.schedule_details",
-            this.get("order.id")
-          );
-        });
+      const orderId = this.get("order.id");
+      const goodcityRequests = this.get("goodcityRequests");
+      try {
+        await this.runTask(
+          Promise.all(
+            goodcityRequests.map(async (gr, index) => {
+              const params = {
+                packageTypeId: gr.packageType.get("id"),
+                quantity: gr.quantity,
+                description: gr.description
+              };
+              if (!gr.id) {
+                return this.createGoodsDetails(orderId, params, index);
+              } else {
+                params.id = gr.id;
+                return this.updateGoodsDetails(orderId, params);
+              }
+            })
+          )
+        );
+        this.transitionToRoute("order.schedule_details", orderId);
+      } catch (e) {
+        this.get("messageBox").alert(e.responseJSON.errors[0].message);
```

Comment:
Since this is multiple promises, I guess if there are multiple errors you are catching only the first ones.